de.metas.adempiere.adempiere/base/src/main/java/de/metas/phonecall/PhonecallSchemaVersionId.java

```diff
@@ -0,0 +1,81 @@
+package de.metas.phonecall;
+
+import javax.annotation.Nullable;
+
+import de.metas.util.Check;
+import de.metas.util.lang.RepoIdAware;
+import lombok.NonNull;
+import lombok.Value;
+
+/*
+ * #%L
+ * de.metas.adempiere.adempiere.base
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+@Value
+public class PhonecallSchemaVersionId implements RepoIdAware
+{
+	int repoId;
+
+	@NonNull
+	PhonecallSchemaId phonecallSchemaId;
+
+	public static PhonecallSchemaVersionId ofRepoId(@NonNull final PhonecallSchemaId phonecallSchemaId, final int phonecallSchemaVersionId)
```

Comment:
please consider if and why you need the `PhonecallSchemaId` in this class

Reply:
it's OK from my POV..might be a good idea in this case, but e.g. at `org.adempiere.warehouse.LocatorId` I felt this approach to be a big nuisance.
On the other hand, for the phonecall model, we can have one repo class to create them, very unlike from the 100 places where we create `LocatorId`s..

Reply:
Talked with Teo again and he advised to keep this PhonecallSchemaId ( similar in other places)  because this PhonecallSchemaVersionId shall never exist outside of the PhonecallSchemaId  and we should make this hierarchy obvious.